src/electron/platform/android/appium-service-configurator.ts

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import ADB from 'appium-adb';
+import {
+    AndroidServiceConfigurator,
+    DeviceInfo,
+    PackageInfo,
+    PermissionInfo,
+} from 'electron/platform/android/android-service-configurator';
+
+type AdbDevice = {
+    udid: string;
+};
+
+const servicePackageName: string = 'com.microsoft.accessibilityinsightsforandroidservice';
+
+export class AppiumServiceConfigurator implements AndroidServiceConfigurator {
+    private readonly adb: ADB;
+
+    constructor(adb: ADB) {
+        this.adb = adb;
+    }
+
+    public getConnectedDevices = (): Promise<Array<DeviceInfo>> => {
+        return new Promise<Array<DeviceInfo>>(async (resolve, reject) => {
```

Comment:
There's a few places in this file where you're using a pattern like

```ts
public someAsyncMethod = (): Promise<OutputT> => {
    return new Promise<OutputT>(async (resolve, reject) => {
        try {
            const output = await doWork();
            resolve(output);
        } catch (error) {
            reject(error);
        }
    });
}
```

You can simplify this a lot by just making the method itself async; the language can deal with most of the wrapping you're currently doing by hand:

```ts
public someAsyncMethod = async (): Promise<OutputT> => {
    return await doWork();
}
```

Reply:
Thanks! Much nicer in this format :)